daemons/controld/controld_te_events.c

```diff
@@ -465,40 +470,39 @@ process_graph_event(xmlNode *event, const char *event_node)
 
     if (status == PCMK_LRM_OP_INVALID) {
         // We couldn't attempt the action
-        crm_info("Transition %d action %d (%s on %s): %s",
-                 transition_num, action_num, id, uname,
-                 services_lrm_status_str(status));
+        crm_notice("Received result operation %s on %s: %s (rc=%d) "
+                   CRM_XS " Transition %d action %d target-rc=%d call-id=%d",
+                   id, uname, services_lrm_status_str(status), rc,
+                   transition_num, action_num, target_rc, callid);
```

Comment:
A couple of minor suggestions: use "result of operation", and move rc=%d after CRM_XS.

Everything before CRM_XS will show up in both the system log and pacemaker.log, while everything after CRM_XS will show up only in pacemaker.log. We try to keep the system log to the minimum information needed to understand what happened, with more technical information such as numeric codes only in pacemaker.log.

I realize there is a challenge in that native English speakers will likely find the readable string more understandable than the numeric code, while non-native English speakers might find the opposite. But I think in principle, I'd rather keep the system log readable, since a user can search pacemaker.log for the same message if they need more detail. It would be nice to have translations of log messages at some point, but the work needed to keep them up-to-date would be significant.

Reply:
> A couple of minor suggestions: use "result of operation",

I'll fix it.

> and move rc=%d after CRM_XS. ... I'd rather keep the system log readable ...

I see. (understood the intent of https://github.com/ClusterLabs/pacemaker/commit/67b4104dd820ea527c2d6caeffb8e2266af9dc2c)

However, if a machine monitors (parses) the messages, wouldn't it be easier to have ```(rc=X)```?
I'll reconfirm the user's opinion/intent.

Reply:
> However, if a machine monitors (parses) the messages, wouldn't it be easier to have `(rc=X)`?

The rc will still be available in pacemaker.log, which can be grabbed for parsing.

A cluster is a highly complex system, so Pacemaker has to balance the risk of logging too little information to troubleshoot a complicated problem with the risk of logging so much that users are overwhelmed and can't find what they're looking for. Logging less information to the system log and more information to pacemaker.log is intended to help with this. Users can start with the system log to find the most critical information, and then only if they can't figure it out, they can look at the detailed logs in pacemaker.log.

Reply:
We are advising users to monitor syslog,
 * As you said, since pacemaker.log is an internal log, it may be frequently changed for implementation reasons. Therefore, we believe that the possibility that compatibility will be lost due to version upgrade or application of Errata is greater than syslog.

and we want machine-readable logs.
 * Since the monitoring is done by machine. 
 * I think the latter, with rc output, is less affected by the format change.
```
# grep ".*pacemaker-controld.*notice:.*Result of operation .* on .*: error$" /var/log/messages
# grep ".*pacemaker-controld.*notice:.*Result of operation .* on .*: Timed Out$" /var/log/messages
# grep ".*pacemaker-controld.*notice:.*Result of operation .* on .*: not running$" /var/log/messages
```
```
# grep ".*pacemaker-controld.*notice:.*Result of operation .* on .*: .* (rc=1)" /var/log/messages
# grep ".*pacemaker-controld.*notice:.*Result of operation .* on .*: .* (rc=7)" /var/log/messages
```
However, now that I understand the logging policy as well, if outputting rc to syslog is unacceptable, please review https://github.com/ClusterLabs/pacemaker/pull/2115/commits/c229d19abb1664da5db50602b2fe3be368ae5994.

Reply:
Hi,

pacemaker.log is not really an internal log, but a detail log. The idea is that users should start with the system log, and if that's not sufficient to help them figure out their problem, they can look in the detail log for more verbose information. Hopefully most problems can be resolved with only the system log.

You are correct that the content of the log messages, in both logs, can change significantly from version to version. While this is unfortunate for searching, the readability of the log messages has been one of the biggest complaints from users, so we've spent a lot of time trying to make them easier to understand. I expect that over time, as we make improvements, the content will stabilize and change less frequently.

There is always some tension between making log messages that are easily understood by humans and log messages that are suited to machine parsing. While I understand the benefits of having machines parse the system log, pacemaker approaches the problem by making the system logs as human-friendly as possible and offering the detail log for machine parsing.